src/client/components/Switch.tsx

```diff
@@ -23,31 +23,57 @@ const SwitchNobble = styled(Tick)`
   background-color: ${colorsV3.white};
 `
 
+/**
+ * The idea is to build a custom checkbox component that is accesibible and that
+ * provides a good UX when it comes about controlling it: e.g it should possible
+ * to bind it with a label e toggle it by clicking on the label.
+ * We could solve the accessible part of the problem by using aria-* attributes.
+ * However, the UX part would require a bunch of JS to emmit the events properly.
+ * With that said, one well adopted solution is to use a hidden checkbox to handle
+ * check/unchecked state that's visually hidden along with your custom implmentation
+ * of a checkbox (styles, animation, etc)
+ */
+// Based on https://css-tricks.com/inclusively-hidden/
+const InclusiveHiddenCheckbox = styled.input`
+  clip: rect(0 0 0 0);
+  clip-path: inset(50%);
+  height: 1px;
+  overflow: hidden;
+  position: absolute;
+  white-space: nowrap;
+  width: 1px;
+`
+
 interface SwitchProps {
+  className?: string
   value: boolean
-  onChange: (newValue: boolean) => void
+  onChange: () => void
 }
 
-export const Switch: React.FC<SwitchProps> = ({ value, onChange }) => (
-  <SwitchContainer
-    onTap={() => {
-      onChange(!value)
-    }}
-    animate={value ? 'checked' : 'unchecked'}
-    variants={{
-      checked: {
-        backgroundColor: colorsV3.gray900,
-        borderColor: colorsV3.gray900,
-      },
-      unchecked: {
-        backgroundColor: colorsV3.white,
-        borderColor: colorsV3.gray300,
-      },
-    }}
-    transition={{ duration: 0.1 }}
-    role="checkbox"
-    aria-checked={value}
-  >
-    <SwitchNobble />
-  </SwitchContainer>
+export const Switch: React.FC<SwitchProps> = ({
+  className,
+  value,
+  onChange,
+}) => (
+  <>
+    <InclusiveHiddenCheckbox type="checkbox" onChange={onChange} />
+    <SwitchContainer
+      className={className}
+      aria-hidden
```

Comment:
nit: I would just change to

```suggestion
      aria-hidden={true}
```